core/metacling/src/TClingCallbacks.cxx

```diff
@@ -845,6 +845,13 @@ bool TClingCallbacks::tryResolveAtRuntimeInternal(LookupResult &R, Scope *S) {
       return false;
    }
 
+   // FIX-8367
+   // Do not try to resolve if/switch/while/for statement, if already annotated
+   // for(i=0;i<0;); was failing due to multiple addDecls for the same variable.
```

Comment:
I believe we should not litter the source code with references to bug reports. If there's an explanation for this code that works without, can you maybe edit this comment?